airbyte-integrations/bases/debezium/src/test/java/io/airbyte/integrations/debezium/internals/DebeziumRecordIteratorTest.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.debezium.internals;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.mockito.Mockito.mock;
+
+import io.airbyte.integrations.debezium.CdcTargetPosition;
+import io.debezium.engine.ChangeEvent;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.concurrent.LinkedBlockingQueue;
+import org.apache.kafka.connect.source.SourceRecord;
+import org.junit.jupiter.api.Test;
+
+public class DebeziumRecordIteratorTest {
+
+  @Test
+  public void getHeartbeatPositionTest() {
```

Comment:
We should include more tests for this function:
- what if is null (or not heartbeat event?)
- if the size > 1
- ...
For me that function is quite complex (which makes hard to test it as well).